runtime/lua/vim/lsp.lua

```diff
@@ -1529,6 +1529,48 @@ function lsp.omnifunc(findstart, base)
   return -2
 end
 
+--- Provides an interface between the built-in client and a `formatexpr` function.
+--- Currently only supports a single client.
+--@param start_line 1-indexed line, defaults to |v:lnum|
+--@param end_line 1-indexed line, calculated based on {start_line} and |v:count|
+--@param timeout_ms optional, defaults to 500ms
+function lsp.formatexpr(start_line, end_line, timeout_ms)
+  timeout_ms = timeout_ms or 500
+  print(start_line, end_line)
+
+  if not start_line or not end_line then
+    if vim.fn.mode() == 'i' or vim.fn.mode() == 'R' then
```

Comment:
Does this need to handle `ic` and `ix` as well? (I'm not sure when does modes are active)